datafusion/functions-aggregate-common/src/utils.rs

```diff
@@ -145,8 +148,11 @@ impl<T: DecimalType> DecimalAverager<T> {
         if let Ok(value) = sum.mul_checked(self.target_mul.div_wrapping(self.sum_mul)) {
             let new_value = value.div_wrapping(count);
 
-            let validate =
-                T::validate_decimal_precision(new_value, self.target_precision);
+            let validate = T::validate_decimal_precision(
```

Comment:
Due to  this (get better messages)
- https://github.com/apache/arrow-rs/pull/8396

Reply:
I'll add "Closes #3666" to the PR body 👍